openapi.yaml

```diff
@@ -4711,16 +4711,32 @@ components:
       example: null
       description: |-
         The namespace the `process_id` is valid for.
+
+        The following options are pre-defined by the openEO API, but additional
+        namespaces may be introduced by back-ends or in a future version of the API.
         
-        By default, the namespace is `null`, which includes all pre-defined processes listed at `GET /processes`.
-        The namespace `user` refers to all user-defined processes listed as `GET /process_graphs`.
-        Additional namespaces may be introduced by back-ends or in a future version of the API.
+        * `null` (default): Checks both user-defined and pre-defined processes,
+           but prefers user-defined processes if both are available.
+           This allows users to add missing pre-defined processes for portability,
+           e.g. common processes from [processes.openeo.org](https://processes.openeo.org)
+           that have a process graph included.
+           If a user-defined process is prioritized over a pre-defined process,
+           a log entry SHOULD be added.
+        * `backend`: Uses exclusively the pre-defined processes listed at `GET /processes`.
+        * `user`: Uses exclusively the user-defined processes listed at `GET /process_graphs`.
+
+        If multiple processes with the same identifier exist, Clients SHOULD
+        inform the user that it's recommended to select a namespace.
     process_id:
       type: string
       description: |-
-        Unique identifier for the process.
+        An identifier for the process.
         
-        MUST be unique across its namespace (e.g. pre-defined processes or user-defined processes).
+        The identifier MUST be unique across its namespace (e.g. pre-defined
+        processes or user-defined processes).
+
+        Clients SHOULD warn the user if a user-defined process is added with the 
+        same identifier as one of the pre-defined process.
```

Comment:
This note about "should warn" is under the `namespace` field when describe process graphs right?  Shouldn't it be in the docs of creating a UDF?

Reply:
No, it's defined under the process id field.

Reply:
I mean: the warning should happen on `PUT /process_graph/{}`  , not when evaluating a process graph (for batch jobs/services/synchronous). So I would put the `Clients SHOULD warn ..` closer to the `PUT /process_graph/{}` docs

Reply:
And yes, I'll add it there in addition.